go/parquet/internal/encoding/boolean_encoder.go

```diff
@@ -0,0 +1,112 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package encoding
+
+import (
+	"github.com/apache/arrow/go/arrow/bitutil"
+	"github.com/apache/arrow/go/parquet"
+	"github.com/apache/arrow/go/parquet/internal/utils"
+)
+
+const boolBufSize = 1024
+
+// PlainBooleanEncoder encodes bools as a bitmap as per the Plain Encoding
+type PlainBooleanEncoder struct {
+	encoder
+	nbits      int
+	bitsBuffer []byte
+}
+
+// Type for the PlainBooleanEncoder is parquet.Types.Boolean
+func (PlainBooleanEncoder) Type() parquet.Type {
+	return parquet.Types.Boolean
+}
+
+// Put encodes the contents of in into the underlying data buffer.
+func (enc *PlainBooleanEncoder) Put(in []bool) {
+	if enc.bitsBuffer == nil {
+		enc.bitsBuffer = make([]byte, boolBufSize)
+	}
+
+	bitOffset := 0
+	// first check if we are in the middle of a byte due to previous
+	// encoding of data and finish out that byte's bits.
+	if enc.nbits > 0 {
+		bitsToWrite := utils.MinInt(enc.nbits, len(in))
+		beg := (boolBufSize * 8) - enc.nbits
+		for i, val := range in[:bitsToWrite] {
+			bitmask := uint8(1 << uint((beg+i)%8))
+			if val {
+				enc.bitsBuffer[(beg+i)/8] |= bitmask
+			} else {
+				enc.bitsBuffer[(beg+i)/8] &= bitmask ^ 0xFF
```

Comment:
might just be pre-caffience but isn't this just `~bitmask` (or does go not have that concept

Reply:
huh, you're right, i didn't even realize this was just a bitwise not.

So Go doesn't have a `~` operator, but you can use the XOR operator to do a unary bitwise NOT. So this could just be `^bitmask` i'll fix that.